pkg/resource-managers/topology.go

```diff
@@ -40,23 +42,31 @@ func topStringKeyFunc(obj interface{}) (string, error) {
 	return s, nil
 }
 
-type instanceTopologyManager struct {
-	ec2                 iface.EC2
+// InstanceTopologyManagerInterface enables mocking the InstanceTopologyManager.
+type InstanceTopologyManagerInterface interface {
+	GetNodeTopology(ctx context.Context, instanceType string, region string, instanceID string) (*types.InstanceTopology, error)
+}
+
+// InstanceTopologyManager manages getting instance topology for nodes.
+type InstanceTopologyManager struct {
```

Comment:
does this need to be a public type? I would name the interface this and make the impl private

Reply:
Done